File: Clase6/SoftProgSolution/SoftProgModel/src/main/java/pe/edu/pucp/softprog/ventas/model/OrdenVenta.java
package pe.edu.pucp.softprog.ventas.model;
import pe.edu.pucp.softprog.gestclientes.model.Cliente;
import pe.edu.pucp.softprog.rrhh.model.Empleado;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class OrdenVenta {
    private int idOrdenVenta;
    private Empleado empleado;
    private Cliente cliente;
    private List<LineaOrdenVenta> lineasOrdenVenta;
    private double total;
    private Date fechaHora;
    private boolean activo;

    public OrdenVenta(){
        this.lineasOrdenVenta = new ArrayList<>();
    }

    public int getIdOrdenVenta() {
        return idOrdenVenta;
    }

    public void setIdOrdenVenta(int idOrdenVenta) {
        this.idOrdenVenta = idOrdenVenta;
    }

    public Empleado getEmpleado() {
        return empleado;
    }

    public void setEmpleado(Empleado empleado) {
        this.empleado = empleado;
    }

    public Cliente getCliente() {
        return cliente;
    }

    public void setCliente(Cliente cliente) {
        this.cliente = cliente;
    }

    public List<LineaOrdenVenta> getLineasOrdenVenta() {
        return lineasOrdenVenta;
    }

    public void setLineasOrdenVenta(List<LineaOrdenVenta> lineasOrdenVenta) {
        this.lineasOrdenVenta = lineasOrdenVenta;
    }

    public double getTotal() {
        return total;
    }

    public void setTotal(double total) {
        this.total = total;
    }

    public Date getFechaHora() {
        return fechaHora;
    }

    public void setFechaHora(Date fechaHora) {
        this.fechaHora = fechaHora;
    }

    public boolean isActivo() {
        return activo;
    }

    public void setActivo(boolean activo) {
        this.activo = activo;
    }

    public void calcularSutotalesYTotal(){
        total = 0;
        for(LineaOrdenVenta lov : lineasOrdenVenta) {
            lov.calcularSubtotal();
            total += lov.getSubtotal();
        }
    }

    public void agregarLineaOrdenVenta(LineaOrdenVenta lineaOrdenVenta){
        this.lineasOrdenVenta.add(lineaOrdenVenta);
    }
}

File: Clase12/ProyectosNET/SoftProgSolution/SoftProgModel/Ventas/OrdenVenta.cs
using SoftProgModel.GestClientes;
using SoftProgModel.RRHH;
using System;
using System.Collections.Generic;
using System.Text;

namespace SoftProgModel.Ventas
{
    public class OrdenVenta
    {
        private int _idOrdenVenta;
        private Empleado _empleado;
        private Cliente _cliente;
        private List<LineaOrdenVenta> _lineasOrdenVenta;
        private double _total;
        private DateTime _fecha;
        private bool _activo;

        public OrdenVenta()
        {
            _lineasOrdenVenta = new List<LineaOrdenVenta>();
        }
        public int IdOrdenVenta { get => _idOrdenVenta; set => _idOrdenVenta = value; }
        public Empleado Empleado { get => _empleado; set => _empleado = value; }
        public Cliente Cliente { get => _cliente; set => _cliente = value; }
        public List<LineaOrdenVenta> LineasOrdenVenta { get => _lineasOrdenVenta; set => _lineasOrdenVenta = value; }
        public double Total { get => _total; set => _total = value; }
        public DateTime Fecha { get => _fecha; set => _fecha = value; }
        public bool Activo { get => _activo; set => _activo = value; }
    }
}
